project/JsonToYaml.scala

```diff
@@ -0,0 +1,39 @@
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.dataformat.yaml.{YAMLFactory, YAMLGenerator}
+import com.fasterxml.jackson.module.scala.DefaultScalaModule
+import sbt.Keys._
+import sbt._
+
+import scala.collection.mutable
+import scala.util.Try
+
+object JsonToYaml {
+  val toYaml = taskKey[Unit]("Generate YAML OpenAPI specification from JSON")
+  def settings: Seq[Setting[_]] = Seq(
+    toYaml := {
+
+      val yamlFactory = new YAMLFactory().configure(YAMLGenerator.Feature.WRITE_DOC_START_MARKER, false)
+      val jsonMapper  = new ObjectMapper().registerModule(DefaultScalaModule)
+      val yamlMapper  = new ObjectMapper(yamlFactory).registerModule(DefaultScalaModule)
+
+      val jsonFile: File = baseDirectory.value / "target/swagger/swagger.json"
+      val yamlFile: File = baseDirectory.value / "target/swagger/application.yaml"
+
+      Try {
+        val jsonString = IO.read(jsonFile)
+        val jsonMap    = jsonMapper.readValue(jsonString, classOf[Map[String, Any]])
+
+        val specification = mutable.LinkedHashMap(
+          "openapi"    -> jsonMap("openapi"),
+          "info"       -> (jsonMap("info").asInstanceOf[Map[String, Any]] + ("version" -> version.value.stripSuffix("-SNAPSHOT"))),
+          "tags"       -> jsonMap("tags"),
+          "paths"      -> jsonMap("paths"),
+          "components" -> jsonMap("components")
+        )
+
+        val yamlString = yamlMapper.writeValueAsString(specification)
+        IO.write(yamlFile, yamlString)
```

Comment:
it would be great to delete the old file so we don't have any confusion on which file to use

Reply:
Any existing JSON/YAML are overwritten by the SBT task run as we're using a static path. Do you suggest deleting the recently-generated JSON file?

Reply:
yes, that's what I'm suggesting

Reply:
Updated